### ISSUE TITLE
Support "full name" teammate name display setting

### ISSUE THREAD

**Issue by jtdaugherty** [matterhorn-chat/matterhorn#420]
The server and account settings include a teammate name display preference that has a "full name" option. Right now, Matterhorn does not support that option. This ticket depends on #350.

**Comment by jtdaugherty** [comment]
As per #559: we should also support "acronym" matching of users (e.g. "John Doe" should be matchable with "JD"). Apparently the official client supports this.

**Comment by misery** [comment]
> As per #559: we should also support "acronym" matching of users (e.g. "John Doe" should be matchable with "JD"). Apparently the official client supports this.

Sorry, I mean ```Nickname``` instead of acronym. https://docs.mattermost.com/help/settings/account-settings.html

**Comment by jtdaugherty** [comment]
Oh! Nickname rendering and matching is already supported. If your account preferences (or server preferences) are set to display nicknames instead of usernames, Matterhorn will honor that. Matterhorn's channel selection matching will only match names based on the displayed name, though - so you cannot match full names if nicknames are enabled.

**Comment by misery** [comment]
Oh well, I thought I can search for nickname even if I use username as displayed name.

**Comment by jtdaugherty** [comment]
It's possible to search for users by any string if you use the `@` autocompletion, but when searching the channel list, only what's displayed can be searched. (Otherwise it would lead to a confusing UI: if you could search by nickname when nickname was not displayed, that would be strange.)

For autocompletion, type `@` in the message area. Then type any string at all -- part of a full name, nickname, or username -- and the resulting matches from the server will be shown for tab-completion.

**Comment by pieqq** [comment]
> Oh! Nickname rendering and matching is already supported. If your account preferences (or server preferences) are set to display nicknames instead of usernames, Matterhorn will honor that. Matterhorn's channel selection matching will only match names based on the displayed name, though - so you cannot match full names if nicknames are enabled.

In my Mattermost instance, under `Account Settings > Display`, I have the following options for `Teammate Name Display`:

- Show username
- Show nickname if one exists, otherwise show first and last name
- Show first and last name

I selected `Show first and last name` and saved. This works well in the Webapp.

However, in Matterhorn, I still see the username being used instead of the full name.

Is this option implemented in Matterhorn? Is the current issue about implementing this? Sorry, I'm a bit confused. :)

**Comment by jtdaugherty** [comment]
@pieqq You are right - Matterhorn does not yet support displaying users by full name. It *does* support searching for them in tab-completion by full name, but Matterhorn only supports the first two options that you listed as far as user display is concerned.

**Comment by pieqq** [comment]
Thanks for the clarification! I suppose I don't need to open another issue regarding this, since the current one seems to fit the description :)

Anyway, thanks for Matterhorn. I'm still getting used to the keybindings, but it feels much more responsive than the default Webapp!

**Comment by jtdaugherty** [comment]
@pieqq You are welcome - it is always great to hear that people enjoy using Matterhorn!
